File: week_2/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Prompt the user for some text
    string text = get_string("Text: ");

    // Count the number of letters, words, and sentences in the text
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    // Compute the Coleman-Liau index
    int idx = round(0.0588 * (((float) letters / (float) words)) * 100 -
                    0.296 * (((float) sentences / (float) words)) * 100 - 15.8);
    // Print the grade level
    if (idx >= 1 && idx <= 16)
    {
        printf("Grade %i\n", idx);
    }
    else if (idx > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Before Grade 1\n");
    }
}

int count_letters(string text)
{
    // Return the number of letters in text
    int sum = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalnum(text[i]))
        {
            sum += 1;
        }
    }
    return sum;
}

int count_words(string text)
{
    // Return the number of words in text
    int sum = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isspace(text[i]))
        {
            sum += 1;
        }
    }
    return sum + 1;
}

int count_sentences(string text)
{
    // Return the number of sentences in text
    int sum = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == '!' || text[i] == '.' || text[i] == '?')
        {
            sum += 1;
        }
    }
    return sum;
}

File: week_6/sentimental-readability/readability.py
from cs50 import get_string
import re

def main():
    text = get_string("Text: ")
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    # Compute the Coleman-Liau index
    idx = round(0.0588 * (letters / words * 100) -
                0.296 * (sentences / words * 100) - 15.8, 0)

    if idx >= 1 and idx <= 16:
        print(f"Grade {int(idx)}")
    elif idx > 16:
        print("Grade 16+")
    else:
        print("Before Grade 1")

def count_letters(text):
    sum = 0

    for i in range(0, len(text)):
        if text[i].isalpha():
            sum += 1
    return sum

def count_words(text):
    sum = 0

    for i in range(0, len(text)):
        if text[i] == " ":
            sum += 1
    return sum + 1

def count_sentences(text):
    sum = 0

    for i in range(0, len(text)):
        if text[i] in ["!", ".", "?"]:
            sum += 1
    return sum

if __name__ == "__main__":
    main()
